File: 942.增减字符串匹配.py
#
# @lc app=leetcode.cn id=942 lang=python3
#
# [942] 增减字符串匹配
#

# @lc code=start
class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        length = len(s)
        high = length
        low = 0
        ans = []
        for c in s:
            if(c == 'I'):
                ans.append(low)
                low += 1
            else:
                ans.append(high)
                high -= 1
        ans.append(low)
        return ans
# @lc code=end

File: code/942.增减字符串匹配.cpp
/*
 * @lc app=leetcode.cn id=942 lang=cpp
 *
 * [942] 增减字符串匹配
 */
#include<iostream>
using namespace std;
#include<vector>
#include<string>
// @lc code=start
class Solution {
public:
    vector<int> diStringMatch(string s) {
        vector<int> ans;
        int low = 0, high = s.size();
        for(char c : s)
        {
            if(c == 'I')
            {
                ans.push_back(low);
                low++;
            }
            else
            {
                ans.push_back(high);
                high--;
            }
        }
        ans.push_back(low);
        return ans;
    }
};
// @lc code=end

